### PR TITLE
Add completion entropy metrics

### PR DESCRIPTION
**Pull request by lewtun**
# What does this PR do?

Metric from the DAPO paper: https://dapo-sia.github.io

Useful for measuring diversity / entropy collapse

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: trl/trainer/grpo_trainer.py [hunk 1]
```diff
@@ -815,12 +833,37 @@ def _get_per_token_logps(self, model, input_ids, attention_mask, logits_to_keep,
...
-            # Divide logits by sampling temperature.
-            # See https://huggingface.co/blog/the_n_implementation_details_of_rlhf_with_ppo#policy-training-implementation-details
-            logits = logits / self.temperature
-            logps = selective_log_softmax(logits, input_ids_batch)  # compute logprobs for the input tokens
+
+            # Divide logits by sampling temperature
+            scaled_logits = logits / self.temperature
+
+            # Compute full probability distributions
+            probs = torch.softmax(scaled_logits, dim=-1)  # (B, seq_len, vocab_size)
```
Comment (qgallouedec) [inline trl/trainer/grpo_trainer.py:841]: It seems to me that this would introduce a big memory spike. In fact, this was the reason for using `selective_log_softmax` in the first place, as the memory spike actually occurs here.
What's more, all results (`probs`) are kept in VRAM.

Comment (lewtun) [reply trl/trainer/grpo_trainer.py:841]: Thanks, let me try `selective_log_softmax` and clean up the VRAM

**Comment by lewtun** [comment]
I realised this will also interfere with Liger and is not worth the hassle for just a metric. Closing for now
